### PR TITLE
When signing up a user, send them an e-mail with their password - #93 - BEHROUZ

### PR DESCRIPTION
**Pull request by bkarimii**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,8 +75,8 @@
 		"webpack-merge": "^5.9.0"
 	},
 	"dependencies": {
-		"axios": "^1.7.2",
 		"@babel/runtime": "^7.22.15",
+		"axios": "^1.7.2",
```
Comment (soonick) [inline package.json]: Seems like nothing changed here. Remove this change and the change in package-lock.json from the PR

File: server/mail.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ require("dotenv").config();
 const transporter = nodemailer.createTransport({
 	host: process.env.MAIL_HOST,
 	port: process.env.MAIL_PORT,
-	secure: true,
+	secure: false,
```
Comment (soonick) [inline server/mail.js]: Why did you change this?

File: server/mail.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ const transporter = nodemailer.createTransport({
 
 const sendEmail = async ({ recipient, subject, message }) => {
 	return await transporter.sendMail({
-		from: "no_reply@example.com",
+		from: "no_reply@love-me-tender.com",
```
Comment (soonick) [inline server/mail.js]: Did you test that sending the e-mail works?

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,4 @@
 dist/
 node_modules/
 .DS_Store
+.env
```
Comment (soonick) [inline .gitignore:5]: Update the README with instructions of what should be included in the .env file

**Review by soonick — CHANGES_REQUESTED** [review]

File: server/mail.js (hunk 2 above)
Comment (bkarimii) [reply server/mail.js]: It gives a server error when you set it to true.

File: server/mail.js (hunk 3 above)
Comment (bkarimii) [reply server/mail.js]: the test model works , but not in prod version , I couldn't find a free smtp provider .

File: package.json (hunk 1 above)
Comment (bkarimii) [reply package.json]: I installed Nodemailer and the dotenv package again, but it was just an updated version. Should I revert the packages to the old versions?

Comment (soonick) [reply package.json]: This only change I can see in this file is the axios dependency being moved. I don't see any new dependencies.

File: server/mail.js (hunk 3 above)
Comment (soonick) [reply server/mail.js]: What do you mean by test model?

Comment (bkarimii) [reply server/mail.js]: When I test sending an email with Mailtrap, I see the email in the inbox, but I cannot send an email to a real person because Mailtrap does not provide this service. I tried to use Mailgun and SendGrid, but they didn't have the service either.

Comment (soonick) [reply server/mail.js]: Did you try this? https://edigleyssonsilva.medium.com/how-to-send-emails-securely-using-gmail-and-nodejs-eef757525324 

I added a gmail address to our Bitwarden.

File: .gitignore (hunk 4 above)
Comment (soonick) [reply .gitignore:5]: @bkarimii , Don't forget to do this

**Review by soonick — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -61,3 +61,14 @@ Then run above command again.
...
+## User Sign-Up Password Emailing
+
+To configure the SMTP server, add the following variables to your .env file and set their values based on your provider:
+
+MAIL_HOST=<your_smtp_host>
+MAIL_PORT=<your_smtp_port>
+MAIL_USER=<your_smtp_user>
+MAIL_PASSWORD=<your_smtp_password>
+
+Replace <your_smtp_host>, <your_smtp_port>, <your_smtp_user>, and <your_smtp_password> with the appropriate values provided by your SMTP service.
```
Comment (soonick) [inline README.md:78]: Did you add the values for the account we are using to our Bitwarden bucket?

File: README.md [hunk 6]
```diff
@@ -61,3 +61,14 @@ Then run above command again.
 ```sh
 node ../node_modules/db-migrate/bin/db-migrate create name-for-the-migration --sql-file
 ```
+
+## User Sign-Up Password Emailing
```
Comment (soonick) [inline README.md]: We also need to add this to the .env file:

```
DB_PASSWORD=opensesame
DB_USERNAME=postgres
```

Please add it somewhere in the README

**Review by soonick — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (bkarimii) [reply README.md]: But these are secret values , should I add them to the .env ? although we already had them in the .env file and these values belong to the the DB config.

Comment (soonick) [reply README.md]: This README must have all the steps necessary to run the system locally. Having those values in the `.env` file is a requirement for running the system locally, so it must be mentioned in the `.env` file.

The values I mentioned above are only for the local environment, so we can add those values to the README.

You are right that those values re related to the DB, so they don't belong in this section. We should either have a section specifically for the DB or a single session that talks about all the entries we need in the `.env` file.

File: README.md [hunk 7]
```diff
@@ -61,3 +61,18 @@ Then run above command again.
...
+MAIL_HOST=<your_smtp_host>
+MAIL_PORT=<your_smtp_port>
+MAIL_USER=<your_smtp_user>
+MAIL_PASSWORD=<your_smtp_password>
+
+Replace <your_smtp_host>, <your_smtp_port>, <your_smtp_user>, and <your_smtp_password> with the appropriate values provided by your SMTP service.
+
+To run the database, add the following variables to your .env file:
+DB_PASSWORD=opensesame
+DB_USERNAME=postgres
```
Comment (soonick) [inline README.md]: Create a new section for this, since it's not part of the email configuration

```suggestion
## Local database configuration

To configure your system to connect to the local database, add the following variables to your .env file:

\```
DB_PASSWORD=opensesame
DB_USERNAME=postgres
\```
```

File: README.md [hunk 8]
```diff
@@ -61,3 +61,18 @@ Then run above command again.
 ```sh
 node ../node_modules/db-migrate/bin/db-migrate create name-for-the-migration --sql-file
 ```
+
+## User Sign-Up Password Emailing
```
Comment (soonick) [inline README.md]: This configures the SMTP server for the whole system

```suggestion
## SMTP configuration
```

**Review by soonick — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -62,17 +62,17 @@ Then run above command again.
 node ../node_modules/db-migrate/bin/db-migrate create name-for-the-migration --sql-file
 ```
 
-## User Sign-Up Password Emailing
+## Data base credentials
```
Comment (soonick) [inline README.md:65]: Database is one word

**Review by soonick — APPROVED** [review]
